modules/kinesis/src/main/scala/com/snowplowanalytics/snowplow/enrich/kinesis/Sink.scala

```diff
@@ -224,6 +222,50 @@ object Sink {
       prre
     }
 
+  /**
+   * The result of trying to write a batch to kinesis
+   *  @param toRetry Records to retry, either because of throttling or an internal error
+   *  @param wasThrottled Whether at least one of retries is because of throttling
+   *  @param exampleInternalError A message to help with logging
+   */
+  private case class TryBatchResult(
+    toRetry: Vector[PutRecordsRequestEntry],
+    wasThrottled: Boolean,
+    exampleInternalError: Option[String]
+  ) {
+    def wasSuccessful: Boolean =
```

Comment:
It might be either for the understanding of the code to rename this as `shouldRetry` and then have `wasSuccessfull = ! shouldRetry`